lib/mayaUsd/fileio/writeJobContext.h

```diff
@@ -134,6 +135,12 @@ class UsdMayaWriteJobContext
         const VtVec3fArray& bbox,
         const UsdTimeCode&  timeSample);
 
+    /// Used to cache the animated blend shape weight plugs that need to be sampled per-frame.
+    /// This is cached at the writeJob-level because the state needs to persist across instances of
+    /// the meshWriter->Write() function.
+    MAYAUSD_CORE_PUBLIC
+    MPlugArray mBlendShapesAnimWeightPlugs;
```

Comment:
It's rather surprising to see translator-specific data in this generic context. What do others think? Should we make it easier to attach any blind data to the context instead of hardcoding something specific each time a translator needs it?

Reply:
Yes; I foresee a lot of future needs for this where a translator will need to retain state between invocations or similar, especially for certain types of Maya nodes.

Reply:
@kxl-adsk Based on our discussion, I've gone ahead and made the change to move this into the meshWriter translator instead, along with updating the unit test as also discussed. Please let me know if there are other questions/concerns.

Reply:
Sounds good. I re-requested the review from @williamkrick and dismissed the stale review since it was no longer applicable based on what we discussed. I also started a new PF.